### DISCUSSION TITLE
Electron support

### DISCUSSION THREAD

**Discussion by thruflo** [electric-sql/electric#545] [Feature requests]
Currently Electron doesn't support ES modules. This appears to be a blocker to using Electric with it.

ESM is in nightly: https://github.com/electron/electron/pull/37535#issuecomment-1705721459 and there's this workaround https://github.com/mifi/build-electron

We should get Electron working and publish an example.

**Comment by aquarius-wing** [comment]
Maybe you should use this [electron-vite](https://electron-vite.org/guide/#scaffolding-your-first-electron-vite-project) to build your electron app

**Reply by CooperPair** [reply]
@aquarius-wing would it be possible for you to share any example the way you have integrated electric sql with electron app ?

**Reply by aquarius-wing** [reply]
I will give it a try.Please wait

**Comment by barbalex** [comment]
Electron 28 now supports esm: https://www.electronjs.org/blog/electron-28-0

**Comment by mcmadafly** [comment]
I've got @electric-sql/pglite working as part of the electron start process but I don't know how to connect my client (Vue) to access the db?

The .env (in vue - using the example app) points to ELECTRIC_SERVICE and ELECTRIC_PG_PROXY_PORT but I'm not sure what these values are or where the pglite lives.
```
t {
  debug: 0,
  dataDir: undefined,
  fsType: 'memoryfs',
  waitReady: Promise { <pending> }
}
{
  rows: [ { message: 'Hello world' } ],
  fields: [ { name: 'message', dataTypeID: 25 } ],
  affectedRows: 0
}
```

**Comment by MJRT** [comment]
That may need write a new [driver](https://electric-sql.com/docs/integrations/drivers/other/generic) (in render process) through [IPC](https://www.electronjs.org/docs/latest/tutorial/ipc) to access db (in main process). 
I may try it out in a while.

**Comment by KyleAMathews** [comment]
👋 we've been working the last month on a rebuild of the Electric server over at a temporary repo https://github.com/electric-sql/electric-next/

You can read more about why we made the decision at https://next.electric-sql.com/about

We're really excited about all the new possibilities the new server brings and we hope you'll check it out soon and give us your feedback.

We're now moving the temporary repo back here. As part of that migration we're closing all the old issues, PRs, and discussions. We really appreciate you taking the time to start this discussion! If it's still relevant in the new system, please reopen.

**Comment by marc-shapiro** [comment]
Kyle, why am I getting hundreds of github notifications from you all of a sudden?  Looks like an infinite loop.

				Marc
